### PR TITLE
Improve the pytest sanity check

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)
This PR improved the pytest sanity check.

* Add recovery support in sanity check
* Make sanity check items configurable from test scripts
* Add retry support of checking services and interfaces status in 
  case the DUT was just rebooted or performed config reload.
* Make sanity check easier to be extended in the future.

### Type of change

- [] Bug fix
- [x] Testbed and Framework(new/improvement)
- [] Test case(new/improvement)

#### How did you do it?
Deprecated the old `base_sanity` fixture and created a more feature rich sanity_check plugin.

#### How did you verify/test it?
Test run cases with this sanity check plugin.

### Documentation 

https://github.com/wangxin/sonic-mgmt/blob/pytest-sanity-check-pr/tests/common/plugins/sanity_check/README.md

### CODE REVIEW

File: tests/common/plugins/sanity_check/checks.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+def check_interfaces(dut):
+    logger.info("Checking interfaces status...")
+
+    networking_uptime = dut.get_networking_uptime().seconds
+    timeout = max((SYSTEM_STABILIZE_MAX_TIME - networking_uptime), 0)
+    interval = 20
+    logger.info("networking_uptime=%d seconds, timeout=%d seconds, interval=%d seconds" % \
+                (networking_uptime, timeout, interval))
+
+    mg_facts = dut.minigraph_facts(host=dut.hostname)['ansible_facts']
```
Comment (mykolaf) [inline tests/common/plugins/sanity_check/checks.py]: Could we use config_facts instead of minigraph_facts?

Comment (wangxin) [reply tests/common/plugins/sanity_check/checks.py]: Yes, we can. But could you please elaborate why? Are we deprecating minigraph_facts?

Comment (mykolaf) [reply tests/common/plugins/sanity_check/checks.py]: Short answer: due to BSL testing

If we configure switch with 'l2' mode with 
`sonic-cfggen -H -p -k  $HWSKU --preset l2`
only the config_db will contain new expected config, and the minigraph will be outdated

Comment (wangxin) [reply tests/common/plugins/sanity_check/checks.py]: OK, I’ll change it to use config_facts.

Comment (wangxin) [reply tests/common/plugins/sanity_check/checks.py]: Just pushed a commit replacing minigraph_facts with config_facts.

**Comment by yxieca** [comment]
@wangxin can you solve the merge conflicts?

**Comment by wangxin** [comment]
@yxieca Sure

**Comment by yxieca** [comment]
@wangxin can you resolve the merge conflict again? You can do "git rebase -i github/master" and then force push your branch to your fork. That way extra changes in master branch won't show.

**Comment by wangxin** [comment]
@yxieca Thanks for the tip. Resolved conflicts.

**Comment by wangxin** [comment]
Trying to toggle ports to recover before rebooting or config reloading is a good idea, I'll add this to sanity check in the future. Thanks!

**Pull request merged**